gui/main.py

```diff
@@ -0,0 +1,248 @@
+import sys 
+
+from PyQt5.QtWidgets import *
+from PyQt5.QtCore import *
+from PyQt5.QtGui import *
+import time
+
+
+import socket
+import subprocess
+import sys
+from datetime import datetime
+import json
+
+from scanner_thread import split_processing
+
+class MainWindow(QMainWindow):
+    label = None
+    scanInProgress = False
+
+    def __init__(self, *args, **kwargs):
+        super(MainWindow, self).__init__(*args, **kwargs)
+
+        self.setWindowTitle("Port Scanner")
+        self.initUI()
+        
+        
+
+    def initUI(self):
+
+        font = QFont()
+        font.setFamily("Consolas")
+        font.setPointSize(12)
+        font.setBold(False)
+        font.setWeight(50)
+        
+        self.scanBtn = QPushButton("Scan", self)
+        self.scanBtn.setGeometry(QRect(280, 180, 75, 23))
+        self.scanBtn.clicked.connect(self.scanButtonClicked)
+
+        self.hostText = QTextEdit(self)
+        self.hostText.setGeometry(QRect(330, 41, 151, 31))
+
+        self.hostLabel = QLabel("Host Address", self)
+        self.hostLabel.setGeometry(QRect(180, 40, 121, 31))
+        self.hostLabel.setFont(font)
+
+        self.lowerRangeLabel = QLabel("Lower Range", self)
+        self.lowerRangeLabel.setGeometry(QRect(70, 120, 121, 31))
+        self.lowerRangeLabel.setFont(font)
+
+        self.higherRangeLabel = QLabel("Higher Range", self)
+        self.higherRangeLabel.setGeometry(QRect(340, 120, 121, 31))
+        self.higherRangeLabel.setFont(font)
+
+        self.portRangeLabel = QLabel("Port Range", self)
+        self.portRangeLabel.setGeometry(QRect(50, 80, 121, 31))
+        self.portRangeLabel.setFont(font)
+
+        self.dividerLine = QFrame(self)
+        self.dividerLine.setGeometry(QRect(175, 90, 391, 16))
+        self.dividerLine.setFrameShape(QFrame.HLine)
+        self.dividerLine.setFrameShadow(QFrame.Sunken)
+
+        self.lowRangeSpinner = QSpinBox(self)
+        self.lowRangeSpinner.setMaximum(65535)
+        self.lowRangeSpinner.setGeometry(QRect(180, 120, 91, 31))
+
+        self.highRangeSpinner = QSpinBox(self)
+        self.highRangeSpinner.setMaximum(65535)
+        self.highRangeSpinner.setGeometry(QRect(460, 120, 91, 31))
+
+        self.createProgressBar()
+
+        self.scanResultsLabel = QLabel("Scan Results",self)
+        self.scanResultsLabel.setGeometry(QRect(270, 250, 121, 31))
+        self.scanResultsLabel.setFont(font)
+
+        self.resultTextField = QPlainTextEdit(self)
+        self.resultTextField.setGeometry(QRect(35, 290, 570, 150))
+        self.resultTextField.setReadOnly(True)
+
+        self.statusBar().showMessage('Ready')
+        self.statusBar().setStyleSheet("background-color : grey")
+
```

Comment:
@Sohaib03 Extra new lines?

Reply:
I followed the PEP8 guideline for new lines. If I remove these lines, it will reduce readability.

Reply:
However, it seems I have added multiple lines in some places. I will remove them.

Reply:
Yes, but remove the extra lines.